vyper/cli/vyper_json.py

```
@@ -254,6 +254,8 @@ def get_settings(input_dict: dict) -> Settings:
 
     optimize = input_dict["settings"].get("optimize")
     experimental_codegen = input_dict["settings"].get("experimentalCodegen")
+    if experimental_codegen is None:
+        experimental_codegen = input_dict["settings"].get("venom")
```

Maybe throw an exception if both are set